bob-backend/src/pearl/utils.rs

```diff
@@ -112,14 +112,15 @@ impl Utils {
             return time.timestamp().try_into().unwrap();
         }
         let mut start_time = match period {
-            period if period <= ChronoDuration::days(1) => time.date().and_hms(0, 0, 0),
+            period if period <= ChronoDuration::days(1) => time.date_naive().and_hms_opt(0, 0, 0).unwrap(),
             period if period <= ChronoDuration::weeks(1) => {
-                let time = time.date().and_hms(0, 0, 0);
+                let time = time.date_naive().and_hms_opt(0, 0, 0).unwrap();
                 time - ChronoDuration::days(i64::from(time.weekday().num_days_from_monday() - 1))
             }
             _ => panic!("pearid: {} is too large", period),
         };
 
+        let time = time.naive_utc();
```

Comment:
You should convert `DateTime<Utc>` to `NaiveDateTime` earlier (right before `start_time` assignment). Currently it is not safe, because `date_naive` use local time according to the source code (which is still `UTC`, but depends on function argument), while `naive_utc` explicitly use UTC time. Furthermore, to preserve original behavior all calculations should be performed in local time, so `naive_local()` function should be used